consul/api@v1.9.0 uses go1.16-only variable

#### Overview of the Issue

When pulling the latest `consul/api` (`v1.9.0`) I get a build error when using go < 1.16 due to the use of `io.Discard`

#### Reproduction Steps

```
% go version
go version go1.14.15 darwin/amd64
% go get github.com/hashicorp/consul/api@latest
go: github.com/hashicorp/consul/api latest => v1.9.0
# github.com/hashicorp/consul/api
../../../../../../go/pkg/mod/github.com/hashicorp/consul/api@v1.9.0/api.go:1072:17: undefined: io.Discard
```

#### Fix

The [`go.mod`](https://github.com/hashicorp/consul/blob/master/api/go.mod) file says that go1.12 is the minimum version.
To keep `consul/api` compatible with previous versions of Go, it should use `ioutil.Discard` rather than `io.Discard`, which was not introduced until `go1.16`

--- a/api/api.go
+++ b/api/api.go
@@ -1069,7 +1069,7 @@ func requireOK(d time.Duration, resp *http.Response, e error) (time.Duration, *h
 // is necessary to ensure that the http.Client's underlying RoundTripper is able
 // to re-use the TCP connection. See godoc on net/http.Client.Do.
 func closeResponseBody(resp *http.Response) error {
-	_, _ = io.Copy(io.Discard, resp.Body)
+	_, _ = io.Copy(ioutil.Discard, resp.Body)
 	return resp.Body.Close()
 }
 

--- a/command/agent/agent.go
+++ b/command/agent/agent.go
@@ -4,7 +4,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
-	"io"
+	"io/ioutil"
 	"os"
 	"os/signal"
 	"path/filepath"
@@ -182,7 +182,7 @@ func (c *cmd) run(args []string) int {
 	if config.Logging.LogJSON {
 		// Hide all non-error output when JSON logging is enabled.
 		ui.Ui = &cli.BasicUI{
-			BasicUi: mcli.BasicUi{ErrorWriter: c.ui.Stderr(), Writer: io.Discard},
+			BasicUi: mcli.BasicUi{ErrorWriter: c.ui.Stderr(), Writer: ioutil.Discard},
 		}
 	}
 

